jena-arq/src/main/java/org/apache/jena/riot/RDFParserBuilder.java

```diff
@@ -504,8 +504,10 @@ private HttpClient buildHttpClient() {
             Header header = new BasicHeader(k, v);
             hdrs.add(header);
         });
-        HttpClient hc = CachingHttpClientBuilder.create().setDefaultHeaders(hdrs).build();
-        return hc;
+        HttpClient hc = HttpOp.createPoolingHttpClientBuilder()
```

Comment:
Should we maybe just put the default headers into the clients built by `HttpOp` as options (e.g. `HttpOp.createPoolingHttpClientBuilder(boolean useDefaultAcceptHeaderForRdf)` or something like that?